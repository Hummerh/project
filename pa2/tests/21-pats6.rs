enum Colors {
      Red,
      Gold,
      Violet
}

fn main() {
      match (Colors::Red) {
         Colors::Red => { },
         Colors::Gold => { },
         Colors::Violet => { }
      }
}
