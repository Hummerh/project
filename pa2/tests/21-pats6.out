(crate
    (items
        (enum-def
            (id
                (Colors))
            (enum-ctor-defs
                (enum-ctor-def
                    (id
                        (Red)))
                (enum-ctor-def
                    (id
                        (Gold)))
                (enum-ctor-def
                    (id
                        (Violet)))))
        (fn-def
            (id
                (main))
            (block
                (match
                    (enum
                        (enum-ctor
                            (id
                                (Colors))
                            (id
                                (Red))))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Colors))
                                        (id
                                            (Red)))))
                            (block))
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Colors))
                                        (id
                                            (Gold)))))
                            (block))
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Colors))
                                        (id
                                            (Violet)))))
                            (block))))))))